### PR TITLE
chore: run on ci type checking and unit tests

### PR DESCRIPTION
**Pull request by teebszet**
# 🔘 PR Type
- [x] Bugfix
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no API changes)

# 📜 Background
https://linear.app/xverseapp/issue/ENG-3327/fix-all-tsc-errors-on-web-extension

depends on https://github.com/secretkeylabs/xverse-core/pull/296

# 🔄 Changes
- fix: fix all simple type errors
- chore: replaced all direct imports of xverse-core internals, to stop declaring type errors on those
- chore: commented on and ignored remaining type errors on existing code, which looks complex to change the code behaviour
- chore: turn on tsc --noEmit in CI checks
- chore: turn off transpileOnly in webpack build config
- chore: update ts-loader, webpack, and use fork-ts-checker-webpack-plugin to speed up type checking (from ~30s -> ~10s initial run)

- fix: bumped xverse-core to 3.1.1 with fixes 
https://github.com/secretkeylabs/xverse-core/releases/tag/v3.1.1

# 🖼 Screenshot / 📹 Video
Include screenshots or a video demonstrating the changes. This is especially helpful for UI changes.

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Comment by teebszet** [comment]
@victorkirov @m-aboelenein

there's 1 remaining type error importing a missing (deprecated?) type from sats-connect.
please see if that can be removed and fixed

**Comment by teebszet** [comment]
found it: https://github.com/secretkeylabs/sats-connect/commit/f9cdd4066f62ee90c825c51f339c35bdd89a698c

File: src/common/types/message-types.ts [hunk 1]
```diff
@@ -125,7 +129,7 @@ export type SignPsbtResponseMessage = Message<
   ExternalSatsMethods.signPsbtResponse,
   {
     signPsbtRequest: string;
-    signPsbtResponse: SignPsbtResponse | string;
+    signPsbtResponse: SignTransactionResponse | string;
```
Comment (teebszet) [inline src/common/types/message-types.ts:132]: https://github.com/secretkeylabs/sats-connect/commit/f9cdd4066f62ee90c825c51f339c35bdd89a698c

File: src/app/screens/ledger/importLedgerAccount/index.tsx [hunk 2]
```diff
@@ -256,7 +256,11 @@ function ImportLedger(): JSX.Element {
       setCurrentStep(ImportLedgerSteps.ADD_ADDRESS);
       setIsButtonDisabled(true);
       if (isBitcoinSelected) {
-        const { btcCreds, ordinalsCreds, newAccountId } = await importBtcAccounts(true);
+        const importedBtcAccounts = await importBtcAccounts(true);
+        if (!importedBtcAccounts) {
+          throw new Error('No accounts');
+        }
```
Comment (teebszet) [inline src/app/screens/ledger/importLedgerAccount/index.tsx:262]: would previously throw a "Cannot destructure Property of Undefined" error

File: src/app/screens/ledger/importLedgerAccount/index.tsx [hunk 3]
```diff
@@ -298,10 +302,11 @@ function ImportLedger(): JSX.Element {
...
       handleClickNext();
       if (isBitcoinSelected) {
-        const { btcCreds, ordinalsCreds, newAccountId } = await importBtcAccounts(
-          true,
-          masterFingerPrint,
-        );
+        const importedBtcAccounts = await importBtcAccounts(true, masterFingerPrint);
+        if (!importedBtcAccounts) {
+          throw new Error('No accounts');
+        }
```
Comment (teebszet) [inline src/app/screens/ledger/importLedgerAccount/index.tsx:308]: would previously throw a "Cannot destructure Property of Undefined" error

File: src/app/screens/swap/useSwap.test.ts [hunk 4]
```diff
@@ -1,64 +1,7 @@
...
-        to: newCurrency,
-        from: updateOppositeCurrencyIfSameAsSelected(state, { newCurrency, side }),
-      };
-    default:
-      return state;
-  }
-};
+import { describe, expect, test } from 'vitest';
+import { Side } from './types';
+import { selectedTokenReducer } from './useSwap';
```
Comment (teebszet) [inline src/app/screens/swap/useSwap.test.ts:4]: https://linear.app/xverseapp/issue/ENG-2474/fix-useswap-test-file-and-jest-unit-tests-in-general

**Review by victorkirov — DISMISSED** [review]
👏 🎉 🚀 💪 
Thank youuuu!!!!! 😄

**Comment by DuskaT021** [comment]
@teebszet tested can be merged

**Pull request merged**